telegram/ext/_applicationbuilder.py

```diff
@@ -928,6 +931,42 @@ async def post_init(application: Application) -> None:
         self._post_init = post_init
         return self
 
+    def post_shutdown(
+        self: BuilderType, post_shutdown: Callable[[Application], Coroutine[Any, Any, None]]
+    ) -> BuilderType:
+        """
+        Sets a callback to be executed by :meth:`Application.run_polling` and
+        :meth:`Application.run_webhook` *after* executing :meth:`Application.shutdown` but
+        *before* executing :meth:`Updater.shutdown` respectively.
+
+        Tip:
+            This can be used for custom shutdown logic that requires to await coroutines, e.g.
+            closing a database connection
+
+        Example:
+            .. code::
+
+                async def post_shutdown(application: Application) -> None:
+                    await db_connection.close()
```

Comment:
```suggestion
                    await application.bot_data['database'].close()
```

just an idea - makes it more obvious why we might need the `application` parameter